packages/custom_lint/lib/src/workspace.dart

```diff
@@ -633,39 +645,80 @@ publish_to: 'none'
               .nonNulls
               .toList(),
         ),
-    };
+    }.map(
+      (key, value) {
+        // Any dependency which is in both `dependencies` and `dev_dependencies` is treated as a regular dependency.
+        if (value.regularDependencies.isNotEmpty &&
+            value.devDependencies.isNotEmpty) {
+          value.regularDependencies.addAll(value.devDependencies);
+          value.devDependencies.clear();
+        }
+        return MapEntry(key, value);
```

Comment:
What is the purpose of this?

Reply:
I don't think it's right to pick one or the other. Both can have different constraints
We should respect both constraints, if specified in both dependencies and dev_dependencies.

Reply:
I did it this way due to this comment:

https://github.com/invertase/dart_custom_lint/blob/dcce163a2864bf794f1d39a4744708573ed773ba/packages/custom_lint/lib/src/workspace.dart#L645-L651

It seemed you wanted to avoid writing the dependency twice.

If that doesn't bother you, i'll change it to write both.

Reply:
I think that's out of date. To begin with, it's odd that the comment is commented :P